xonsh/aliases.py

```diff
@@ -329,12 +329,14 @@ def bang_bang(args, stdin=None):
 def which_version():
     """Returns output from system `which -v`"""
     if sys.platform == 'darwin':
-        if ('gnu' not in subprocess.check_output(['which'],
-                                                 stderr=subprocess.PIPE)):
+        try:
```

Comment:
I think that this try-except block should be bumped up to the top level. I am on linux (Ubuntu) but I also don't have a `-v` option:

```console
scopatz@artemis ~ $ /usr/bin/which --version
Illegal option --
Usage: /usr/bin/which [-a] args
```